### PR TITLE
Logging fixes and catches

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Hardens logging against sudden and unexpected cases where we try to log to a category that is not initialized or the file goes missing.

## Why It's Good For The Game

Logging needs to be robust and bullet proof

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
This requires updating of CI to use 515 for compilation to work

**Comment by ZephyrTFA** [comment]
Added a preprocessor if for byond version, instead just makes a do nothing macro with a compiler warning

**Comment by Ghommie** [comment]
> Added a preprocessor if for byond version, instead just makes a do nothing macro with a compiler warning

Which is causing the linters and Windows Build CI to fail.

File: code/modules/logging/log_entry.dm [hunk 1]
```diff
@@ -99,19 +99,37 @@ GENERAL_PROTECT_DATUM(/datum/log_entry)
...
+		}; \
+		in_error_recovery = TRUE; \
+		logger.Log(LOG_CATEGORY_INTERNAL_ERROR, "attempting to perform file error recovery: [file]"); \
+		logger.init_log_category(logger.log_categories[category]?.type); \
+		call(src, UNLINT(__PROC__))(arglist(args)); \
+		return; \
+	}; \
+	in_error_recovery = FALSE;
+#else
+#define CHECK_AND_TRY_FILE_ERROR_RECOVERY(file)
```
Comment (JohnFulpWillard) [inline code/modules/logging/log_entry.dm:118]: couldn't you give it it's old behavior of just crashing

Comment (ZephyrTFA) [reply code/modules/logging/log_entry.dm:118]: we're increasing min version to 515, so no

**Pull request closed without merging**